File: codeforces/virtual/952/C.py
"""
sbhatore
04 August 2024

# Assumptions:
#
#
#
# In:
# 1. T
# 2. n
# 3. a_i, 0 <= i < n
#
# Out:
#
#
#
# Solution:
#
#
#
# pseudocode:
#
# rough-work:
# 1. 1 2 3 4 5
"""

import copy

def calcAnswer():
    pass


def __main__():
    T = int(input())
    for t in range(0, T):
        n = int(input())
        A = list(map(int, input().split()))
        sum = 0
        count = 0
        B = copy.deepcopy(A)
        for i in range(1, n):
            A[i] += A[i - 1]
        #
        for i in range(0, n):
            for j in range(i + 1, n):
                if (B[i] == B[j]):
                    B[j] = -1
        for i in range(0, n):
            for j in range(0, i + 1):
                if ((A[i] - B[j]) == B[j]):
                    count += 1
        #
        print(count)
        #
    #
    return

__main__()

File: codeforces/virtual/952/C.cpp
/*
sbhatore
06 August 2024
*/

#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

int main() {
    int t, T;
    cin >> T;
    for (t = 0; t < T; t += 1) {
        int n;
        vector <long long int> A, B;
        cin >> n;
        for (int i = 0; i < n; i += 1) {
            long long int TEMP;
            cin >> TEMP;
            A.push_back(TEMP);
        }
        for (int i = 0; i < n; i += 1){
            B.push_back(A[i]);
        }
        for (int i = 1; i < n; i += 1) {
            A[i] += A[i - 1];
        }
        //
        for (int i = 0; i < n; i += 1) {
            for (int j = i + 1; j < n; j += 1) {
                if (B[i] == B[j]) {
                    B[j] = -1;
                }
            }
        }
        int count = 0;
        //
        for (int i = 0; i < n; i += 1) {
            for (int j = 0; j < i + 1; j += 1) {
                if ((A[i] - B[j]) == B[j]) {
                    count += 1;
                }
            }
        }
        cout << count << endl;
    }
    return 0;
}
